src/nvim/search.c

```diff
@@ -4347,121 +4348,288 @@ int linewhite(linenr_T lnum)
 }
 
 // Add the search count "[3/19]" to "msgbuf".
-// When "recompute" is true Always recompute the numbers.
-static void search_stat(int dirc, pos_T *pos,
-                        bool show_top_bot_msg, char_u *msgbuf, bool recompute)
+// See update_search_stat() for other arguments.
+static void cmdline_search_stat(int dirc, pos_T *pos, pos_T *cursor_pos,
+                                int show_top_bot_msg, char_u  *msgbuf,
+                                int recompute, int maxcount, long timeout)
```

Comment:
`bool recompute` is correct.

In general, don't change `bool` (or TriState) to `int` unless int values are used.
If int values are used, then always use int values even if Vim uses `FALSE` and `TRUE` for int parameters.

Reply:
changed . + recompute in `update_search_state` changed to `bool`